### ISSUE TITLE
compiler.libcxx is not properly recognized from CMake parameters

### ISSUE THREAD

**Issue by Scarabytes** [conan-io/cmake-conan#229]
I'm using Clang 8 and libc++ on a Linux Mint 18.3 system. I tell CMake via the CC, CXX and CXXFLAGS environment variables (or the CMAKE_* variables) to use Clang and libc++ instead of GCC and libstdc++. However, conan-cmake still seems to prefer libstdc++ over libc++. Am I doing something wrong, or is this maybe a bug?

My conan code (which is run at the very beginning of the CMake project):
``` cmake
if(NOT EXISTS "${CMAKE_BINARY_DIR}/conan.cmake")
  message(STATUS "Downloading conan.cmake from https://github.com/conan-io/cmake-conan/tree/v0.15")
  file(DOWNLOAD "https://github.com/conan-io/cmake-conan/raw/v0.15/conan.cmake"
                "${CMAKE_BINARY_DIR}/conan.cmake"
                STATUS DOWNLOAD_STATUS)

  list(GET DOWNLOAD_STATUS 0 ERRCODE)
  list(GET DOWNLOAD_STATUS 1 MESSAGE)

  if (NOT ${ERRCODE} EQUAL 0)
    message(FATAL_ERROR "Download failed: ${MESSAGE}")
    file(REMOVE "${CMAKE_BINARY_DIR}/conan.cmake")
  else()
    message(STATUS "Downloaded to ${CMAKE_BINARY_DIR}/conan.cmake")
  endif()
endif()

include(${CMAKE_BINARY_DIR}/conan.cmake)

conan_cmake_run(
  CONANFILE ../conanfile.txt
  BUILD missing
  BASIC_SETUP
  CMAKE_TARGETS
)
```

The output when invoking cmake:
Note that I explicitly set the CXX Flags to contain `-stdlib=libc++`. The compiler does use this standard library. Conan however detects `libstdc++11`.
```
$ cmake -DCMAKE_BUILD_TYPE=Debug -DCMAKE_C_COMPILER=clang-8 -DCMAKE_CXX_COMPILER=clang++-8 -DCMAKE_CXX_FLAGS=-stdlib=libc++ ..

-- The CXX compiler identification is Clang 8.0.1
-- Check for working CXX compiler: /usr/bin/clang++-8
-- Check for working CXX compiler: /usr/bin/clang++-8 -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Downloading conan.cmake from https://github.com/conan-io/cmake-conan/tree/v0.15
-- Downloaded to /home/******/Code/ConanProject/cmake-build-debug/conan.cmake
-- Conan: Automatic detection of conan settings from cmake
-- Conan: Settings= -s;build_type=Debug;-s;compiler=clang;-s;compiler.version=8;-s;compiler.libcxx=libstdc++11
-- Conan: checking conan executable
-- Conan: Found program /usr/local/bin/conan
-- Conan: Version found Conan version 1.23.0

-- Conan executing: /usr/local/bin/conan install /home/******/Code/ConanProject/src/../conanfile.txt -s build_type=Debug -s compiler=clang -s compiler.version=8 -s compiler.libcxx=libstdc++11 -g=cmake --build=missing
Configuration:
[settings]
arch=x86_64
arch_build=x86_64
build_type=Debug
compiler=clang
compiler.libcxx=libstdc++11
compiler.version=8
os=Linux
os_build=Linux
[options]
[build_requires]
[env]

---snip---
```

Please let me know if I did something wrong or if you need more information.

**Comment by czoido** [comment]
Hi @Scarabytes,
Thanks a lot for reporting, I have been able to reproduce the issue and it looks like that the error comes from the fact that cmake-conan is not taking into account `CMAKE_CXX_FLAGS` to get the `libcxx` setting. I'll open a PR to fix this. In the meantime, you could provide that flag using  `add_compile_options` or similar and it would be set the right `libcxx`.

**Comment by Scarabytes** [comment]
Thank you so much for looking into this!

**Comment by czoido** [comment]
Fixed by: #197
